### DISCUSSION TITLE
Generate bool type field exception

### DISCUSSION THREAD

**Discussion by yuhaya** [SeaQL/sea-orm#1266] [Q&A] [answered]
## Steps to Reproduce

```rust
ut.table(merchant::Entity).add_column(
            ColumnDef::new(merchant::Column::Online)
                .boolean()
                .not_null()
                .default(false),
        );
```
Generate a data table with bool values

### Expected Behavior

The Online field type in the generated data table should be TINYINT UNSIGNED

### Actual Behavior

The Online field type in the generated data table is TINYINT

### Reproduces How Often

always 

## Versions

- name = "sea-orm"
- version = "0.10.2"

- name = "sea-orm-migration"
- version = "0.10.2"

## Additional Information

influences  query model to JsonValue

- TINYINT -> query model to JsonValue -> number
- TINYINT UNSIGNED -> query model to JsonValue -> bool

**Comment by billy1624** [comment]
Hey @yuhaya, welcome!! Would you mind tell me which database you're using? From what you described, I guess it's MySQL. In MySQL, `BOOLEAN` datatyoe is represented as `TINYINT` https://dev.mysql.com/doc/refman/8.0/en/other-vendor-data-types.html So, this is the expected behaviour.

**Comment by yuhaya** [comment]
> it's MySQL
```rust
ut.table(merchant::Entity).add_column(
            ColumnDef::new(merchant::Column::Online)
                .boolean()
                .not_null()
                .default(false),
        );
```

query model call **into_json()**, The online field will be parsed into a number type

```rust
ut.table(merchant::Entity).add_column(
            ColumnDef::new(merchant::Column::Online)
                .tiny_unsigned()
                .not_null()
                .default(false),
        );
```
query model call **into_json()**, The online field will be parsed into a boolean type

### Expected Behavior

boolean() ->query into_json() -> parsed into boolean
tiny_unsigned() ->query  into_json() -> parsed into number

I guess the problem is in the serde crate, or there is a bug in the into_json method

**Comment by billy1624** [comment, marked as answer]
The `into_json()` method convert `sqlx` query result into `serde_json::Value`. SeaORM rely on the type info provided by `sqlx` to distinguish the correct datatype of each column in the query result.

https://github.com/SeaQL/sea-orm/blob/d32e1eb05859461c49941af3be4219a810d8e6b4/src/query/json.rs#L30-L37

If you look at https://docs.rs/sqlx-core/0.6.2/src/sqlx_core/mysql/types/bool.rs.html#10. You will find that a MySQL `TINYINT UNSIGNED` column correspond to a `bool` in Rust.

Hence, the column defined with datatype `TINYINT UNSIGNED` yield a boolean json value when calling `into_json()` method.

```rs
ut.table(merchant::Entity).add_column(
    ColumnDef::new(merchant::Column::Online)
        .tiny_unsigned()
        .not_null()
        .default(false),
);
```

**Comment by billy1624** [comment]
I think this issues is more of a Q&A instead of bug report

**Comment by yuhaya** [comment]
> The method convert query result into . SeaORM rely on the type info provided by to distinguish the correct datatype of each column in the query result.`into_json()``sqlx``serde_json::Value``sqlx`
> 
> https://github.com/SeaQL/sea-orm/blob/d32e1eb05859461c49941af3be4219a810d8e6b4/src/query/json.rs#L30-L37
> 
> If you look at https://docs.rs/sqlx-core/0.6.2/src/sqlx_core/mysql/types/bool.rs.html#10. You will find that a MySQL column correspond to a in Rust.`TINYINT UNSIGNED``bool`
> 
> Hence, the column defined with datatype yield a boolean json value when calling method.`TINYINT UNSIGNED``into_json()`
> 
> ```rust
> ut.table(merchant::Entity).add_column(
>     ColumnDef::new(merchant::Column::Online)
>         .tiny_unsigned()
>         .not_null()
>         .default(false),
> );
> ```

Thanks, I understand the cause of the problem

**Comment by billy1624** [comment]
No problem! I'll convert this into a Q&A
